File: GeneratedEntityReaderClasses/Components/CollisionTriggerComponent.cs
namespace GeneratedNoitaClasses;
public class CollisionTriggerComponent : NoitaComponentBase,  iNoitaType<CollisionTriggerComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaFloat width { get; set; } = new();
	public NoitaFloat height { get; set; } = new();
	public NoitaFloat radius { get; set; } = new();
	public NoitaString required_tag { get; set; } = new();
	public NoitaBool remove_component_when_triggered { get; set; } = new();
	public NoitaBool destroy_this_entity_when_triggered { get; set; } = new();
	public NoitaInt timer_for_destruction { get; set; } = new();
	public NoitaBool self_trigger { get; set; } = new();
	public NoitaInt skip_self_frames { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		width.Read(s);
		height.Read(s);
		radius.Read(s);
		required_tag.Read(s);
		remove_component_when_triggered.Read(s);
		destroy_this_entity_when_triggered.Read(s);
		timer_for_destruction.Read(s);
		self_trigger.Read(s);
		skip_self_frames.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		width.Write(s);
		height.Write(s);
		radius.Write(s);
		required_tag.Write(s);
		remove_component_when_triggered.Write(s);
		destroy_this_entity_when_triggered.Write(s);
		timer_for_destruction.Write(s);
		self_trigger.Write(s);
		skip_self_frames.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/CollisionTriggerComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaFloat
from BasicNoitaTypes import NoitaInt

from NoitaStream import NoitaStream
@dataclass
class CollisionTriggerComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    width:NoitaFloat = field(default_factory=NoitaFloat)
    height:NoitaFloat = field(default_factory=NoitaFloat)
    radius:NoitaFloat = field(default_factory=NoitaFloat)
    required_tag:NoitaString = field(default_factory=NoitaString)
    remove_component_when_triggered:NoitaBool = field(default_factory=NoitaBool)
    destroy_this_entity_when_triggered:NoitaBool = field(default_factory=NoitaBool)
    timer_for_destruction:NoitaInt = field(default_factory=NoitaInt)
    self_trigger:NoitaBool = field(default_factory=NoitaBool)
    skip_self_frames:NoitaInt = field(default_factory=NoitaInt)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.width.Read(s)
        self.height.Read(s)
        self.radius.Read(s)
        self.required_tag.Read(s)
        self.remove_component_when_triggered.Read(s)
        self.destroy_this_entity_when_triggered.Read(s)
        self.timer_for_destruction.Read(s)
        self.self_trigger.Read(s)
        self.skip_self_frames.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.width.Write(s)
        self.height.Write(s)
        self.radius.Write(s)
        self.required_tag.Write(s)
        self.remove_component_when_triggered.Write(s)
        self.destroy_this_entity_when_triggered.Write(s)
        self.timer_for_destruction.Write(s)
        self.self_trigger.Write(s)
        self.skip_self_frames.Write(s)

